### DISCUSSION TITLE
Create model with uncertainty (unkown paramters that can be updated online)

### DISCUSSION THREAD

**Discussion by Hongyu-ZHOU** [stack-of-tasks/pinocchio#2163] [General]
Hi, 

I am trying to create a model of a quadruped robot from urdf file, and use that model for an adaptive control algorithm. In that case, is it possible to add uncertainty in the model generated? Essentially, I want to have a nominal model generated from the urdf file, augmented with some unknown parts (a vector of known basis function times a vector of unknown parameters.).

Thank you very much.

**Comment by ManifoldFR** [comment]
Hi, I moved this issue to the discussion tab if you don't mind!

I think what you can do is load up your model from the URDF file with some nominal parameters. This will yield a `pinocchio.Model` object in Python (`pinocchio::Model` in C++).

Then, you can write your own logic to play with the parameters you want to change in the `Model` object - gear ratio, friction, damping, inertias, joint placements, gravity. You can do that by accessing them on the object itself - we don't have pre-made functions for this, it's all left to the user.
You can keep using the same `Data` object when changing these properties. Make sure your simulator and/or control loop are pointing to these properties in the `Model` object for the physics.

One of the things I'm not sure you can change easily is the _size_ of the vectors holding parameters, along with the structure of kinematic tree itself (adding/removing joints) and the operational frames. Then, you'd need to work with multiple `Model` instances and switch between them.

Good luck, have fun!
